sdk/python/feast/infra/online_stores/snowflake.py

```diff
@@ -0,0 +1,232 @@
+from feast.repo_config import FeastConfigBaseModel, RepoConfig
+from feast.usage import log_exceptions_and_usage
+
+
+class SnowflakeOnlineStoreConfig(FeastConfigBaseModel):
+    """ Online store config for Snowflake """
+
+    type: Literal["snowflake.online"] = "snowflake.online"
+    """ Online store type selector"""
+
+    config_path: Optional[str] = (
+        Path(os.environ["HOME"]) / ".snowsql/config"
+    ).__str__()
+    """ Snowflake config path -- absolute path required (Can't use ~)"""
+
+    account: Optional[str] = None
+    """ Snowflake deployment identifier -- drop .snowflakecomputing.com"""
+
+    user: Optional[str] = None
+    """ Snowflake user name """
+
+    password: Optional[str] = None
+    """ Snowflake password """
+
+    role: Optional[str] = None
+    """ Snowflake role name"""
+
+    warehouse: Optional[str] = None
+    """ Snowflake warehouse name """
+
+    database: Optional[str] = None
+    """ Snowflake database name """
+
+    schema_: Optional[str] = Field("PUBLIC", alias="schema")
+    """ Snowflake schema name """
+
+    class Config:
+        allow_population_by_field_name = True
+
+
+class SnowflakeOnlineStore(OnlineStore):
+    @log_exceptions_and_usage(online_store="snowflake")
+    def online_write_batch(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        data: List[
+            Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        assert isinstance(config.online_store, SnowflakeOnlineStoreConfig)
+
+        dfs = [None] * len(data)
+        for i, (entity_key, values, timestamp, created_ts) in enumerate(data):
+
+            df = pd.DataFrame(
+                columns=[
+                    "entity_feature_key",
+                    "entity_key",
+                    "feature_name",
+                    "value",
+                    "event_ts",
+                    "created_ts",
+                ],
+                index=range(0, len(values)),
+            )
+
+            timestamp = _to_naive_utc(timestamp)
+            if created_ts is not None:
+                created_ts = _to_naive_utc(created_ts)
+
+            for j, (feature_name, val) in enumerate(values.items()):
+                df.loc[j, "entity_feature_key"] = serialize_entity_key(
+                    entity_key
+                ) + bytes(feature_name, encoding="utf-8")
+                df.loc[j, "entity_key"] = serialize_entity_key(entity_key)
+                df.loc[j, "feature_name"] = feature_name
+                df.loc[j, "value"] = val.SerializeToString()
+                df.loc[j, "event_ts"] = timestamp
+                df.loc[j, "created_ts"] = created_ts
+
+            dfs[i] = df
+            if progress:
+                progress(1)
+
+        if dfs:
+            agg_df = pd.concat(dfs)
+
+            with get_snowflake_conn(config.online_store, autocommit=False) as conn:
```

Comment:
assuming that autocommit=False is ok since we're not doing a transaction? Might be worth adding comments on this